libs/libfpgabitstream/src/bitstream_manager.cpp

```diff
@@ -296,6 +299,79 @@ void BitstreamManager::add_output_net_id_to_block(
   block_output_net_ids_[block] = output_net_id;
 }
 
+void BitstreamManager::overwrite_bitstream(const std::string& path,
+                                           const bool& value) {
+  bool bad_format = true;
+  size_t index = path.rfind("[");
+  std::string bit_string = "";
+  if (index != std::string::npos && path[path.size() - 1] == ']') {
```

Comment:
If you need to check the port, you may use the PortParser

https://github.com/lnis-uofu/OpenFPGA/blob/7f9dffec8994b474e98f815b9b5ffc42fa98e842/libs/libopenfpgautil/src/openfpga_port_parser.h#L24-L40

- Use BasicPort.is_valid() to check if there is any format issue.
- Use BasicPort.get_name() to get the last block name
- Use BasicPort.get_lsb() or get_msb() to get the index for the bit in a leaf block

Reply:
I don't think I want to use the PortParser:
- The parser is too advanced to use in the simple case that I need - https://github.com/lnis-uofu/OpenFPGA/blob/master/libs/libopenfpgautil/src/openfpga_port_parser.cpp#L49-L92. So support so many (5) possibles, then the code is complicated as well. My code is fast and simple
- The most important it do the string to int conversion without checking the string - https://github.com/lnis-uofu/OpenFPGA/blob/master/libs/libopenfpgautil/src/openfpga_port_parser.cpp#L79
   - Not sure if you are aware of if you call std::stoi("10a"), the API will return 10 without any error. You can try it
   - I do not like it. That's why in my code I did make sure every single character is digit because calling std::stoi()

Reply:
@chungshien I understand your points. The port parser may be an overkill but it is designed to handle a wide range of scenarios. This is the motivation of utilities: general and reusable. 

- If you see the utility is not good enough, we can absolutely discuss how to improve it, rather than starting from scratches. 
- In a growing codebase like OpenFPGA, utility functions are strongly needed and should be used on higher layers. As such, developers can quickly implement new features without focusing on detailed parser.
- More codes indicate more risks in maintanence. It becomes worse when there are multiple tools in the framework. When working with the VTR project, we always ensure to re-use existing utility functions.

Reply:
Update Port Parser via https://github.com/chungshien/OpenFPGA/commit/766df0a1b5bae0af7342c55850be9b62fe7d051a